cmd/datasource-token-refresh/main_test.go

```diff
@@ -0,0 +1,171 @@
+				},
+			},
+		},
+		{
+			name: "OK with editing headers",
+			input: grafana.DataSource{
+				URL:  "http://localhost:9090",
+				Type: "prometheus",
+				JSONData: map[string]interface{}{
+					"httpHeaderName1":   "X-Custom-Header",
+					"httpHeaderName2":   "Authorization",
+					"httpHeaderName3":   "X-Custom-Header3",
+					"httpMethod":        "POST",
+					"prometheusType":    "Prometheus",
+					"prometheusVersion": "2.37.0",
+				},
+			},
+			want: grafana.DataSource{
+				URL:  "http://localhost:9090",
+				Type: "prometheus",
+				JSONData: map[string]interface{}{
+					"httpHeaderName1":   "X-Custom-Header",
+					"httpHeaderName2":   "Authorization",
+					"httpHeaderName3":   "X-Custom-Header3",
+					"httpMethod":        "GET",
+					"prometheusType":    "Prometheus",
+					"prometheusVersion": "2.40.0",
+				},
+				SecureJSONData: map[string]interface{}{
+					"httpHeaderValue2": "Bearer 12345",
+				},
+			},
+		},
+		{
+			name:                "prometheus server url override",
+			prometheusServerURL: "http://monitoring.googleapis.com/test",
+			input: grafana.DataSource{
+				Type: "prometheus",
+				URL:  "http://localhost:9090",
+				JSONData: map[string]interface{}{
+					"httpHeaderName1":   "X-Custom-Header",
+					"httpHeaderName2":   "X-Custom-Header2",
+					"httpHeaderName3":   "Authorization",
+					"httpMethod":        "POST",
+					"prometheusType":    "Prometheus",
+					"prometheusVersion": "2.37.0",
+				},
+			},
+			want: grafana.DataSource{
+				URL:  "http://monitoring.googleapis.com/test",
+				Type: "prometheus",
+				JSONData: map[string]interface{}{
+					"httpHeaderName1":   "X-Custom-Header",
+					"httpHeaderName2":   "X-Custom-Header2",
+					"httpHeaderName3":   "Authorization",
+					"httpMethod":        "GET",
+					"prometheusType":    "Prometheus",
+					"prometheusVersion": "2.40.0",
+				},
+				SecureJSONData: map[string]interface{}{
+					"httpHeaderValue3": "Bearer 12345",
+				},
+			},
+		},
+		{
+			name: "prometheus server url not set",
+			input: grafana.DataSource{
+				Type: "prometheus",
+				JSONData: map[string]interface{}{
+					"httpHeaderName1":   "X-Custom-Header",
+					"httpHeaderName2":   "Authorization",
+					"httpHeaderName3":   "X-Custom-Header3",
+					"httpMethod":        "POST",
+					"prometheusType":    "Prometheus",
+					"prometheusVersion": "2.37.0",
+				},
+			},
+			fail: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			*prometheusEndpoint = tt.prometheusServerURL
+			got, err := buildUpdateDataSourceRequest(tt.input, accessToken)
+			if tt.fail {
+				if err == nil {
+					t.Fatalf("unexpectedly succeded")
+				}
+				return
+			}
+			gotJson, err := json.Marshal(got)
+			if err != nil {
+				t.Fatalf("unmarshall gotJson failed with error: %v", err)
+			}
+			wantJson, err := json.Marshal(tt.want)
+			if err != nil {
+				t.Fatalf("unmarshall wantJson failed with error: %v", err)
```

Comment:
```suggestion
				t.Fatalf("unmarshal wantJson failed with error: %v", err)
```

I think it's one `l`?